dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/ajax/ContentletAjax.java

```diff
@@ -851,209 +853,226 @@ else if( fieldbcontentname.startsWith("date") ){
-	                			&& actionlet.getActionlet().getClass().getCanonicalName().equals(PushPublishActionlet.class.getCanonicalName())){
-	                		hasPushPublishActionlet = true;
-	                	}
-	                }
-	                wfActionMap.put( "hasPushPublishActionlet", hasPushPublishActionlet );
-	                try {
-						wfActionMap.put( "wfActionNameStr", LanguageUtil.get( currentUser, action.getName() ) );
-					} catch (LanguageException e) {
-						Logger.error( this, "Could not load language key : " + action.getName() );
-					}
-	                wfActionMapList.add( wfActionMap );
+					JSONObject wfActionMap = new JSONObject();
+					try {
+						wfActionMap.put("name", action.getName());
+						wfActionMap.put("id", action.getId());
+						wfActionMap.put("icon", action.getIcon());
+						wfActionMap.put("assignable", action.isAssignable());
+						wfActionMap.put("commentable", action.isCommentable() || UtilMethods.isSet(action.getCondition()));
+						wfActionMap.put("requiresCheckout", action.requiresCheckout());
+
+						List<WorkflowActionClass> actionlets = APILocator.getWorkflowAPI().findActionClasses(action);
+						for (WorkflowActionClass actionlet : actionlets) {
+							if (actionlet.getActionlet() != null
+									&& actionlet.getActionlet().getClass().getCanonicalName().equals(PushPublishActionlet.class.getCanonicalName())) {
+								hasPushPublishActionlet = true;
+							}
+						}
+						wfActionMap.put("hasPushPublishActionlet", hasPushPublishActionlet);
+						try {
+							wfActionMap.put("wfActionNameStr", LanguageUtil.get(currentUser, action.getName()));
+						} catch (LanguageException e) {
+							Logger.error(this, "Could not load language key : " + action.getName());
+						}
+						wfActionMapList.add(wfActionMap);
 
-                } catch (JSONException e1) {
-					Logger.error(this,  "Could not put property in JSONObject");
+					} catch (JSONException e1) {
+						Logger.error(this, "Could not put property in JSONObject");
+					}
 				}
-            }
 
 
-			searchResult.put( "wfActionMapList", wfActionMapList.toString() );
+				searchResult.put("wfActionMapList", wfActionMapList.toString());
 
-            // End Workflow Actions
+				// End Workflow Actions
 
-			//searchResult.put("structureName", st.getVelocityVarName());
-			Long LanguageId=con.getLanguageId();
-			searchResult.put("languageId", LanguageId.toString());
-			searchResult.put("permissions", permissionsSt.toString());
+				//searchResult.put("structureName", st.getVelocityVarName());
+				Long LanguageId = con.getLanguageId();
+				searchResult.put("languageId", LanguageId.toString());
+				searchResult.put("permissions", permissionsSt.toString());
+			} catch (DotSecurityException e) {
 
-			results.add(searchResult);
+				Logger.debug(this, "Does not have permissions to read the content: " + searchResult, e);
+				searchResult = null;
+
+			} catch (Exception e) {

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)